### PR TITLE
Update from upstream

### PR DESCRIPTION
**Pull request by claravanstaden**
Changes to our crates on the upstream polkadot-sdk.

Polkadot-sdk companion: https://github.com/Snowfork/polkadot-sdk/pull/103

### CODE REVIEW

File: parachain/Cargo.toml [hunk 1]
```diff
@@ -21,14 +21,32 @@ members = [
...
+option-map-unit-fn = { level = "allow", priority = 2 }               # stylistic
+bind_instead_of_map = { level = "allow", priority = 2 }              # stylistic
+erasing_op = { level = "allow", priority = 2 }                       # E.g. 0 * DOLLARS
+eq_op = { level = "allow", priority = 2 }                            # In tests we test equality.
+while_immutable_condition = { level = "allow", priority = 2 }        # false positives
+needless_option_as_deref = { level = "allow", priority = 2 }         # false positives
+derivable_impls = { level = "allow", priority = 2 }                  # false positives
+stable_sort_primitive = { level = "allow", priority = 2 }            # prefer stable sort
+extra-unused-type-parameters = { level = "allow", priority = 2 }     # stylistic
+default_constructed_unit_structs = { level = "allow", priority = 2 } # stylistic
```
Comment (claravanstaden) [inline parachain/Cargo.toml:52]: Same lints as the polkadot-sdk.

File: parachain/pallets/inbound-queue/src/mock.rs [hunk 2]
```diff
@@ -21,8 +21,8 @@ use sp_runtime::{
 	BuildStorage, FixedU128, MultiSignature,
 };
 use sp_std::convert::From;
-use xcm::v3::{prelude::*, MultiAssets, SendXcm};
-use xcm_executor::Assets;
+use xcm::v4::{prelude::*, SendXcm};
```
Comment (yrong) [inline parachain/pallets/inbound-queue/src/mock.rs:24]: Seems the explicit `v4` is unnecessary here, it's by default exposed in `xcm::prelude`

File: parachain/primitives/router/src/outbound/mod.rs [hunk 3]
```diff
@@ -16,7 +16,7 @@ use snowbridge_core::{
 };
 use sp_core::{H160, H256};
 use sp_std::{iter::Peekable, marker::PhantomData, prelude::*};
-use xcm::v3::prelude::*;
+use xcm::v4::prelude::*;
```
Comment (yrong) [inline parachain/primitives/router/src/outbound/mod.rs:19]: same as https://github.com/Snowfork/snowbridge/pull/1123/files#r1462962331

**Review by yrong — APPROVED** [review]
+1

**Review by vgeddes — APPROVED** [review]
+1

File: parachain/pallets/inbound-queue/src/mock.rs (hunk 2 above)
Comment (claravanstaden) [reply parachain/pallets/inbound-queue/src/mock.rs]: Good point! Changed in https://github.com/Snowfork/snowbridge/pull/1123/commits/b8515e6155e4fedd2c7842c6fcd84856c3a197fc

**Pull request merged**